### DISCUSSION TITLE
searchBar event onReturn

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13617] [Q&A] [answered]
To refresh the content of my listView with respect to a search string, I use the event onReturn. It works as expected. Only if searchBar is empty, then the event is not triggered. In consequence, I can't go back to initial status of the list. 

And If I click the searchIcon in the searchBar it also does not trigger the onReturn event at all. The event is only triggered by the searchIcon of the keyboard. Maybe this is specific behavior of the emulator. 

view.xml:
```
<Collection src="test" />
...
<SearchBar id="search" showCancel="true" onReturn="searchOnReturn" />
...

```
controller.js
```

function searchOnReturn(event){
  let pattern = event.value;
  if(typeof pattern === "undefined") pattern = "";
  Alloy.Collections.test.fetch({query: "SELECT * FROM test WHERE title LIKE '" + pattern + "%'"});
}
```

Btw: How to color the underscore, canel- and searchIcon of the searchBar?

**Comment by m1ga** [comment, marked as answer]
I looks like it is not possible because of native restictions: https://stackoverflow.com/a/31573689/5193915
> The queryTextListener wont fire on an empty string event

They do the same as I would do in Titanium since you want to do some more styling: use a own textfield. Create something like this:
```xml
<View>
  <TextField/>
  <ImageView id="search_icon"/>
  <ImageView id="dismiss_icon"/>
</View>
```
and attach your events.

![Screenshot_20221022-212239](https://user-images.githubusercontent.com/4334997/197358839-ea6dffe9-3f21-4fa5-af2f-4c44cc9467de.png)

100% freedom for your look & feel.

**Reply by beppo-ivel** [reply]
And it is also not possible to change the color of the icons in the default searchBar? How do you changed the color of the cursor in your text fields?

**Reply by m1ga** [reply]
Cursor color: I've set `colorPrimary` and `colorAccent` in my theme to orange. You can also try this one https://stackoverflow.com/a/15431107/5193915
With this setup you can change each icon as you want. Use an Label with an icon font instead of a ImageView.

If you still want to use the searchbar you'll have to look at native Android theme examples. E.g. https://stackoverflow.com/a/57735080/5193915 uses a setColorFilter. You could use Hyperloop with this example https://github.com/tidev/titanium_mobile/discussions/13423 and adapt that for the searchbar. 
But as described earlier: way to much work. Create the XML above. Cross-platform and works the same way with more control
